LanguageFeatures/Augmentation-libraries/augmenting_members_A02_t01_lib.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// A non-writable variable declaration is augmented with a setter.
+///
+/// @description Checks that it is a compile-time error if an augmenting
+/// declaration uses `augmented` when the augmented declaration has no concrete
+/// implementation.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_members_A02_t01.dart';
+
+/**/augment void set topLevelVariable(String _) {}
+//  ^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+augment class C {
+  augment static void set staticVariable(String _) {}
+//^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment void set instanceVariable(String _) {}
+//^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+augment mixin M {
+  augment static void set staticVariable(String _) {}
+//^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment void set instanceVariable(String _) {}
+//^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+augment enum E {
+  augment e0;
+  augment static void set staticVariable(String _) {}
+//^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment void set instanceVariable(String _) {}
+//^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+augment extension Ext {
+  augment static void set staticVariable(String _) {}
+//^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+augment extension type ET {
```

Comment:
Here we could also add a setter that matches up with the representation variable.

Reply:
Added.

Reply:
Thanks!